driver/src/main/java/org/neo4j/driver/Config.java

```diff
@@ -149,6 +163,16 @@ public ConfigBuilder withMinIdleTimeBeforeConnectionTest( long milliSecond )
         }
 
         /**
+         * Enable TLS in all connections with server
+         * @param value
+         */
+        public ConfigBuilder withTLSEnabled( boolean value )
```

Comment:
We've learned that irritating users is detrimental to adoption and can even increase friction with our allies, like driver authors. There is always a balance between security and convenience. 

First-trust is a nice alternative to manually propagating certificates. That's you shouting a password to me across a room filled with people we both trust, which I can then take with me to use in a setting where I don't trust the neighbors. More concretely, I can connect to Neo4j while I'm at work with little fuss, then take my laptop home and still feel comfortable that I'm not compromising business information.

That seems pretty nice. 

Equally convenient would be no-TLS, but then I'd have no benefits. Less convenient would be full-TLS, but that'd be most secure. 

Turning off TLS could have lots of useful scenarios, including a client-side which does not have write permission to a file system (which seems to be a requirement), and happens to not care about security.